app/controllers/password_resets_controller.rb

```diff
@@ -0,0 +1,72 @@
+# frozen_string_literal: true
+
+class PasswordResetsController < ApplicationController
+  before_action :load_user_to_update, :valid_user, :check_expiration, only: %i(edit update)
+  before_action :load_user_to_create, only: :create
+  before_action :check_password_presence, only: :update
+
+  def new; end
+
+  def edit; end
+
+  def create
+    @user.create_reset_digest
+    @user.send_password_reset_email
+    flash[:info] = t("layouts.messages.email_instructions")
+    redirect_to root_url
+  end
+
+  def update
+    if @user.update(user_params)
+      log_in(@user)
+      @user.update_column :reset_digest, nil
+      flash[:success] = t("layouts.messages.password_update_success")
+      redirect_to @user
+    else
+      render :edit
+    end
+  end
+
+  private
+
+  def user_params
+    params.require(:user).permit :password, :password_confirmation
+  end
+
+  def load_user_to_update
+    @user = User.find_by email: params[:email]
+    return if @user
+
+    flash[:danger] = t "layouts.messages.user_not_found"
+    render :new
+  end
+
+  def load_user_to_create
+    @user = User.find_by email: params.dig(:password_reset, :email).downcase
+    return if @user
+
+    flash[:danger] = t "layouts.messages.user_not_found"
+    render :new
+  end
```

Comment:
```suggestion
  def load_user
    @user = User.find_by email: (params[:email] || params.dig(:password_reset, :email).downcase)
    return if @user

    flash[:danger] = t "layouts.messages.user_not_found"
    render :new
  end
```
cần j tạo thêm method